cookbooks/aws-parallelcluster-computefleet/test/controls/custom_parallelcluster_node_spec.rb

```diff
@@ -0,0 +1,28 @@
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+control 'custom_parallelcluster_node_installed' do
+  file_cache_path = "/tmp/kitchen/cache"
+
+  title "custom aws-parallelcluster-node should have been installed in the node virtualenv"
+  only_if { !os_properties.redhat_ubi? }
+
+  # Unless we fix the version of aws-parallelcluster-node to be installed, we cannot really say from pip if a custom
+  # node package was installed. The only thing we can test is that the custom node recipe was triggered, which can
+  # be verified from some artifacts left in the kitchen cache folder.
```

Comment:
[Nitpick] As said in a comment for the `install/node.rb` ..if we store the `node_package_url` as part of the config we should be able to check if we are using a custom node version.

Reply:
Yep. I tend to agree, but I cannot spend much more time on this at the moment. We will do it as part of the cookbook refactoring.